src/components/cardModal/CardModal.js

```diff
@@ -0,0 +1,141 @@
+import styles from "@/styles/cardModal.module.scss";
+import { useState, useContext } from "react";
+import { LoggedInContext } from "@/pages/_app";
+
+
+async function sendCardInfo(userName, cardInfo)
+{
+    const resp = await fetch('/api/users/addcard', {
+        "method": "PATCH",
+        "headers": {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        "body": JSON.stringify({
+            "userName": userName,
+            "paymentMethods": cardInfo
+        })
+    })
+    const data = await resp.json();
+
+    return data;
+}
+
+
+
+export default function CardModal2({ showCardModal, handleCloseCardModal, handleCard })
```

Comment:
Callback properties are always (with very few exceptions) given names that start with `on` in React, e.g. `onCloseCardModal` or `onCard`.

It's often the case that people use the `handleCard` naming pattern for the **function passed as the value**, but never for the name of the property. I.e. like this:

```tsx
function handleCard() {
  // Do something
}

<CardModal onCard={handleCard}/>
```

It's important to understand the difference between the name of the property and the value of the property, which may be a named function.